comp/logs/agent/config/config_keys.go

```diff
@@ -109,7 +109,15 @@ func (l *LogsConfigKeys) devModeUseProto() bool {
 	return l.getConfig().GetBool(l.getConfigKey("dev_mode_use_proto"))
 }
 
+func (l *LogsConfigKeys) compressionKind() string {
+	return l.getConfig().GetString(l.getConfigKey("compression_kind"))
+}
+
 func (l *LogsConfigKeys) compressionLevel() int {
+	if l.compressionKind() == "zstd" {
+		return l.getConfig().GetInt(l.getConfigKey("zstd_compression_level"))
```

Comment:
Is it worth having two separate configuration fields?
I would expect the `compression_level` configuration field to apply to any kind of compressor I'm using.

It's not a strong opinion.

Reply:
We need different default values for each compression kind. `gzip` would be 6, `zstd` would be 1. I think the clearest way to do this is to have separate fields.